### PR TITLE
documentation/8260-Dylan-WhatsNewEncouragedUpdateDocumentation

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Added documentation for Encouraged Update and Whats New

## Screenshots/Video
N/A

## Testing
N/A

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
Add documentation for Whats New and Encouraged Update

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Encouraged Update
+ 
+## Feature Summary
+
+Encouraged update displays to the veteran that there is a newer version in the app store to encourgage them to update with a button to click to either download in App update(Android) or to go to the app store to download(iOS). In doing so they will download the latest version of the app and have access to all of the benefits that come with it. We also gave them the option of skipping an update for a particular version which hides this alert until the next version is released. This alert takes priority over the Whats New alert.
+
+This is potentially a precurssor to forced upgrading in the future after so many versions of the app have gone by without upgrading. Currently there is no plan to have a forced upgrade feature, but the ability is there.
```
Comment (dumathane) [inline VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md:7]:
```suggestion
This is potentially a precursor to forced upgrading in the future after so many versions of the app have gone by without upgrading. Currently there is no plan to have a forced upgrade feature, but the ability is there.
```

**Review by dumathane — DISMISSED** [review]

File: VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+This is potentially a precursor to forced upgrading in the future after so many versions of the app have gone by without upgrading. Currently there is no plan to have a forced upgrade feature, but the ability is there.
+
+## Use Cases
+
+* Use Case 1: The version on the device is older than what is in the store so it displays the encouraged update alert(see screenshot)
+* Use Case 2: The version on the device is older than what is in the store and the veteran decided to skip encourage update for this version so it does not display the alert
+* Use Case 3: The version on the device is the same version in the store or newer(updates from the app store role out to devices periodically so it is possible that they updated by the store api is returning to us a older version) so it displays the [Whats New](../WhatsNew/WhatsNew.md) alert if applicable
```
Comment (theodur) [inline VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md:13]:
```suggestion
* Use Case 3: The version on the device is the same version in the store or newer (updates from the app store roll out to devices periodically, so it is possible that the app was updated but the store's API is returning an older version) so it displays the [What's New](../WhatsNew/WhatsNew.md) alert if applicable 
```

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Whats New
```
Comment (theodur) [inline VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:1]:
```suggestion
# What's New
```

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+## Feature Summary
+
+Whats new displays to the veteran what is new in the version that they upgraded to. This alert only shows if the the version of the app the user is on is the most recent in the store and if there is an important enough change to annouce with the provided notes to display to the user. The user can dismiss this alert from reappearing for the current version.
```
Comment (theodur) [inline VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:5]:
```suggestion
What's new displays to the veteran what is new in the version that they upgraded to. This alert is only shown if the version of the app the user is on is the most recent in the store and if there is asignificant change to announce with the provided notes to display to the user. The user can dismiss this alert from reappearing for the current version.
```

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+## Use Cases
+
+* Use Case 1: The user is on the most recent version available in the store and there is notes for the alert to display resulting in the alert displaying(see screenshot)
```
Comment (theodur) [inline VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:9]:
```suggestion
* Use Case 1: The user is on the most recent version available in the store, and there are notes for the alert to display, resulting in the alert being shown (see screenshot)
```

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+* Use Case 2: The user is on the most recent version with notes to display, but the user has dismissed the alert causing it to no longer be displayed
```
Comment (theodur) [inline VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:10]:
```suggestion
* Use Case 2: The user is on the most recent version with notes to display, but the user has dismissed the alert, causing it to no longer be displayed
```

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+* Use Case 3: The user is on a older version of the app and instead sees [Encouraged Update](../EncouragedUpdate/EncouragedUpdate.md)
```
Comment (theodur) [inline VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:11]:
```suggestion
* Use Case 3: The user is on an older version of the app and instead sees [Encouraged Update](../EncouragedUpdate/EncouragedUpdate.md)
```

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+* Use Case 3: The user is on the most recent version but there are no notes to be displayed so there is no alert
```
Comment (theodur) [inline VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:12]:
```suggestion
* Use Case 3: The user is on the most recent version, but there are no notes to be displayed, so there is no alert
```

File: VAMobile/documentation/static/img/encouragedUpdate/EncouragedUpdate.png
Comment (theodur) [inline VAMobile/documentation/static/img/encouragedUpdate/EncouragedUpdate.png:1]: Small nit that's optional, it'd be cleaner if the screenshots weren't displaying the gradient warning. Also, consider cropping out the window header. Same with the other screenshots

**Review by theodur — CHANGES_REQUESTED** [review]
Noticed some typos and made some other suggestions

**Review by theodur — COMMENTED** [review]
Should we also add instructions on how to update the What's New content, or is this just supposed to serve as a briefing of the feature, and that will be added somewhere else?

File: VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Encouraged Update
+ 
+## Feature Summary
+
+Encouraged update displays to the veteran that there is a newer version in the app store to encourgage them to update with a button to click to either download in App update(Android) or to go to the app store to download(iOS). In doing so they will download the latest version of the app and have access to all of the benefits that come with it. We also gave them the option of skipping an update for a particular version which hides this alert until the next version is released. This alert takes priority over the Whats New alert.
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md:5]: added

File: VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+This is potentially a precursor to forced upgrading in the future after so many versions of the app have gone by without upgrading. Currently there is no plan to have a forced upgrade feature, but the ability is there.
+
+## Use Cases
+
+* Use Case 1: The version on the device is older than what is in the store so it displays the encouraged update alert(see screenshot)
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md:11]: added

File: VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+* Use Case 2: The version on the device is older than what is in the store and the veteran decided to skip encourage update for this version so it does not display the alert
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md:12]: added

File: VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/EncouragedUpdate/EncouragedUpdate.md:13]: added

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md (hunk 5 above)
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:9]: added

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md (hunk 7 above)
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:11]: added

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md (hunk 6 above)
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:10]: added

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md (hunk 3 above)
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:1]: added

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md (hunk 4 above)
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:5]: added

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md (hunk 8 above)
Comment (Sparowhawk) [reply VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:12]: added

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+## Feature Summary
+
+What's new displays to the veteran what is new in the version that they upgraded to. This alert is only shown if the version of the app the user is on is the most recent in the store and if there is asignificant change to announce with the provided notes to display to the user. The user can dismiss this alert from reappearing for the current version.
+
+## Use Cases
+
+* Use Case 1: The user is on the most recent version available in the store, and there are notes for the alert to display, resulting in the alert being shown (see screenshot)
+* Use Case 2: The user is on the most recent version with notes to display, but the user has dismissed the alert, causing it to no longer be displayed
+* Use Case 3: The user is on a older version of the app and instead sees [Encouraged Update](../EncouragedUpdate/EncouragedUpdate.md)
+* Use Case 3: The user is on the most recent version, but there are no notes to be displayed, so there is no alert
```
Comment (dumathane) [inline VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:12]:
```suggestion
* Use Case 4: The user is on the most recent version, but there are no notes to be displayed, so there is no alert
```

**Review by dumathane — APPROVED** [review]

File: VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md (hunk 7 above)
Comment (theodur) [reply VAMobile/documentation/docs/App Features/WhatsNew/WhatsNew.md:11]: @Sparowhawk Looks like this suggestion wasn't applied

**Pull request merged**